core/store/src/flat/manager.rs

```diff
@@ -0,0 +1,136 @@
+use crate::flat::{store_helper, POISONED_LOCK_ERR};
+use near_primitives::errors::StorageError;
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::ShardLayout;
+use near_primitives::types::ShardId;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use tracing::debug;
+
+use crate::{Store, StoreUpdate};
+
+use super::chunk_view::FlatStorageChunkView;
+use super::FlatStorage;
+
+/// `FlatStorageManager` provides a way to construct new flat state to pass to new tries.
```

Comment:
This doc-comment has a nice summary sentence, but it fails to leave either an empty space at the end of the line (which forces a `<br />`) or an empty line between the next paragraph, which means that this won't be handled as a proper summary sentence by compliant markdown/commonmark parsers.